model/task_lifecycle.go

```diff
@@ -1451,28 +1451,23 @@ func ClearAndResetStrandedTask(h *host.Host) error {
 		return errors.WithStack(MarkEnd(t, evergreen.MonitorPackage, time.Now(), &t.Details, false))
 	}
 
+	// Only restart failed tasks for stranded tasks.
+	t.SetRestartFailed()
 	return errors.Wrap(ResetTaskOrDisplayTask(t, evergreen.User, evergreen.MonitorPackage, &t.Details), "resetting task")
 }
 
 // ResetTaskOrDisplayTask is a wrapper for TryResetTask that handles execution and display tasks that are restarted
 // from sources separate from marking the task finished. If an execution task, attempts to restart the display task instead.
 // Marks display tasks as reset when finished and then check if it can be reset immediately.
 func ResetTaskOrDisplayTask(t *task.Task, user, origin string, detail *apimodels.TaskEndDetail) error {
-	taskToReset := *t
-	if taskToReset.IsPartOfDisplay() { // if given an execution task, attempt to restart the full display task
-		dt, err := taskToReset.GetDisplayTask()
-		if err != nil {
-			return errors.Wrap(err, "getting display task")
-		}
-		if dt != nil {
-			taskToReset = *dt
-		}
+	if t.IsPartOfDisplay() {
+		return errors.New("cannot restart individual execution task")
```

Comment:
im not sure if this is a non-backwards compatible change but I don't think we should allow anyone to call this function on a execution task

Reply:
You'll need to trace where this function is called to verify an execution task can't be passed in

Reply:
the only place it's called is stranded tasks and the route. im worried about is the reset task route and I'm wondering if any users will complain if we disallow it now

but I'm convinced that it was only written like this to start with to handle stranded tasks and removing it should be fine